azerothcore/mod-solocraft
# Feature and Admin Upgrade/Overhaul

This is a large overhaul of the Solocraft module.

Features and Admin Controls added:
- Groups now work.  Difficulty is now distributed across group size
- Debuff added for groups  trying to exceed the difficulty modifier
- Debuff can be enabled and disabled
- Spellpower is now scaled based on character level and modifier
- Stat modifier can now be changed in config to reel in melee damage
- Individual modification of each dungeon/raid difficulty
- Max level cap to keep high level characters from getting buffed
- Max level cap modifiable in config for each dungeon/raid
- Dungeon/Raid types still used as a catch all if not found as an
individual instance in the config

A database table is included to manage groups and the difficulty
modifier that each group member has applied to them.  It also
deals with server restarts and players having to log out and  log
back in to try to correct stats.  The table is required.

Updated README.md file with current instructions and typo fixes

Fixes #9

Testing:
- Debuff: 
- Character in dungeon with full difficulty modifier applied while not in a group.
- Invite another character to the group.
- Have new group member enter same dungeon.
- Debuff will be applied because of the group member already in the dungeon
- has the full difficulty modifier applied. 
- Have both characters leave dungeon and re-enter it to have the modifier 
- balanced across the current members of the group.     

- Max Level:
- Solocraft.Max.Level.Diff and the Dungeon individual level are added together
- to declare the max level a character can be to get the difficulty modifier 
- applied for the dungeon.
- Use a character whose level is higher than the max level and enter dungeon.
- Character should not receive the difficulty modifier buff for the dungeon.

- Spellpower and Stats modifier adjustments:
- Change modifier in the config.
- Enter dungeon with character and notice stats and/or spellpower numbers
- changed from previous entrance into the dungeon.
